airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -486,8 +489,15 @@ def request_params(
         query = f"SELECT {','.join(selected_properties.keys())} FROM {self.name} "
         if start_date:
             query += f"WHERE {self.cursor_field} >= {start_date} "
+            if next_page_token:
+                primary_key = next_page_token.get("primary_key")
+                if primary_key and self.name not in UNSUPPORTED_FILTERING_STREAMS:
+                    query += f"AND {self.primary_key} > '{primary_key}' "
         if self.name not in UNSUPPORTED_FILTERING_STREAMS:
-            query += f"ORDER BY {self.cursor_field} ASC LIMIT {self.page_size}"
+            order_by_fields = [self.cursor_field]
+            if self.primary_key:
+                order_by_fields.append(self.primary_key)
+            query += f"ORDER BY {','.join(order_by_fields)} ASC LIMIT {self.page_size}"
```

Comment:
Could there be a case where:
date=2 primarykey=2
date=3 primarykey=1
If so, wouldn't we miss the second entry?

Reply:
yes it seems you are right 

it seems we need only add primarykey to filtering if we inside that same date batch

Reply:
OMG things become even more complex

Reply:
Yes, because the order by date is mandatory to make the incremental process work.

Reply:
I'm under the impression this discussion is resolved with the upstream logic. Isn't it?